components/org.wso2.carbon.sp.jobmanager.core/src/main/java/org/wso2/carbon/sp/jobmanager/core/appCreator/SafeZkClient.java

```diff
@@ -28,13 +33,45 @@
  */
 class SafeZkClient {
 
+    private ZkClient zkClient;
+    private ZkUtils zkUtils;
+    private static final Logger log = Logger.getLogger(SafeZkClient.class);
+
     /**
      * Creates a new Zookeeper client for the specified server URL.
      *
-     * @param zooKeeperServerUrl server URL
+     * @param zooKeeperServerUrls server URL
      * @return a new Zookeeper client
      */
-    public ZkClient createZkClient(String zooKeeperServerUrl) {
-        return new ZkClient(zooKeeperServerUrl, 10000, 8000, ZKStringSerializer$.MODULE$);
+    public ZkUtils createZkClient(String[] zooKeeperServerUrls, boolean isSecureKafkaCluster) {
+        for (String zooKeeperServerUrl : zooKeeperServerUrls) {
+            try {
+                zkClient = new ZkClient(zooKeeperServerUrl, 10000, 5000, ZKStringSerializer$.MODULE$);
+                zkUtils = new ZkUtils(zkClient, new ZkConnection(zooKeeperServerUrl), isSecureKafkaCluster);
+                break;
+            } catch (ZkTimeoutException e) {
+                log.error("Zookeeper server at " + zooKeeperServerUrl + " can not be reached.");
```

Comment:
Better to log with 'e' of the ZkTimeoutException

Reply:
Fixed in d93646f